README.md

```diff
@@ -76,6 +76,165 @@ For more info about the API, see the [API reference](https://kontent.ai/learn/re
 
 You can find the Delivery and other SDKs at <https://github.com/Kentico>.
 
+## Used toolchain
+
+This application is based on the [Create React App](https://reactjs.org/docs/create-a-new-react-app.html) using the following template `--template typescript`.
+
+## Model mapping and data fetching
+
+There are two types of model mapping in this application:
+
+### content type -> DTO -> component
+
+Content type definitions are being generated from content types via [Kontent.ai model generator](https://github.com/Kentico/kontent-model-generator-js) tool. All generated types can be found in `src/Models` folder. The `_project.ts` contains information about the project structure such as project languages as well as other structure information like codenames about content types.
+
+### content type -> DTO -> view model -> component
+
+Some models displayed in views might require an adjustment from content types. For example, the `Cafe` content type contains fields for `city` and `street` and we would like to have a model containing an address in the format `city, street`. An example of such a view model is in `CafeModel.tsx` that can be found in the `src/ViewModels` folder. To convert `Cafe` into `CafeModel` the function located in `src/Utilities/CafeListing.ts` can be used.
+
+### Data fetching
+
+This solution fetches data using the [Delivery client](https://github.com/Kentico/kontent-delivery-sdk-js). For more implementation detail to set up the client see `src/Client.ts`. The data are fetched and stored in a `container` component directly in its state. Then they are passed to the `presentation` component. For a better understanding see the code example below. However, depending on your needs, you can use other technologies for managing application states such as:
+
+- [Context](https://reactjs.org/docs/context.html)
+- [Redux](https://react-redux.js.org/)
+- [Flux](https://facebook.github.io/flux/)
+- ...
+
+```tsx
+const Component: React.FC = () => {
+    const [data, setData] = useState([] : DTOComponentProps);
+
+    useEffect(() => {
+      const query = Client.items<GeneratedDTO>()
+          .type(projectModel.contentTypes.generatedDTO.codename)
+          ...
+
+      const items : GeneratedDTO[] = query.ToPromise()

```

Comment:
There should not be an assignment - `setState` does not return anything: https://reactjs.org/docs/react-component.html#setstate